### PR TITLE
add git pre-commit hook

### PR DESCRIPTION
**Pull request by thebino**
## 📑 What does this PR do?

Adds a pre-commit script which can/should run before every commit to make it easier for devs to comply to the project rules and code styles

# ✅ Checklist

- [x] My pull request adheres to the code style of this project
- [x] My code requires changes to the documentation
- [x] I have updated the documentation as required
- [x] All the tests have passed

### CODE REVIEW

File: .githooks/pre-commit [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#!/bin/sh
+#
```
Comment (crocsandcoffee) [inline .githooks/pre-commit:2]: shouldn't all git hook scripts go under .git/hooks ?

File: .githooks/pre-commit [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+# hook script to verify code formatting and runs tests for all modules
+# also it will try to format code and add missing license headers to every kotlin file.
+
+echo "***** Run code formatting and tests ******"
+git stash -q --keep-index
+
+./gradlew check
```
Comment (crocsandcoffee) [inline .githooks/pre-commit:9]: shouldn't we also run `spotlessApply` or `spotlessCheck`?

What does `check` task run?

File: .githooks/pre-commit (hunk 1 above)
Comment (thebino) [reply .githooks/pre-commit:2]: after installing the git hook, you'll find it there. But `.git` is not saved in the git history itself, so it needs to be outside if you want to share it

File: .githooks/pre-commit (hunk 2 above)
Comment (thebino) [reply .githooks/pre-commit:9]: I added `spotlessApply` to the script.
`test` is running `build`, `lint`, `test` and `spotlessCheck` already

**Review by crocsandcoffee — APPROVED** [review]

**Pull request merged**